hal/halx86/acpi/madt.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Source File for MADT Table parsing
+ * COPYRIGHT:   Copyright 2021 Justin Miller <justinmiller100@gmail.com>
+ */
+
+/* INCLUDES *******************************************************************/
+
+#include <hal.h>
+#include <acpi.h>
+/* ACPI_BIOS_ERROR defined in acoutput.h and bugcodes.h */
+#undef ACPI_BIOS_ERROR
+#include <smp.h>
+#define NDEBUG
+#include <debug.h>
+
+/* GLOBALS ********************************************************************/
+
+PROCESSOR_IDENTITY HalpStaticProcessorIdentity[MAXIMUM_PROCESSORS] = {{0}};
+PPROCESSOR_IDENTITY HalpProcessorIdentity = NULL;
+HALP_APIC_INFO_TABLE HalpApicInfoTable;
+ACPI_TABLE_MADT *MadtTable;
+ACPI_SUBTABLE_HEADER *AcpiHeader;
+ACPI_MADT_LOCAL_APIC *LocalApic;
+
+/* FUNCTIONS ******************************************************************/
+
+VOID
+HalpParseApicTables(
+    _In_ PLOADER_PARAMETER_BLOCK LoaderBlock)
+{
+    ULONG_PTR TableEnd;
+    ULONG ValidProcessorCount;
+
+    /* We only support legacy APIC for now, this will be updated in the future */
+    HalpApicInfoTable.ApicMode = 0x10;
+    MadtTable = HalAcpiGetTable(LoaderBlock, 'CIPA');
+
+    AcpiHeader = (ACPI_SUBTABLE_HEADER*)MadtTable;
+    AcpiHeader->Length = sizeof(ACPI_TABLE_MADT);
+    TableEnd = (ULONG_PTR)MadtTable + MadtTable->Header.Length;
+
+    HalpApicInfoTable.ProcessorCount = 0;
+    HalpProcessorIdentity = HalpStaticProcessorIdentity;
+
+    AcpiHeader = (ACPI_SUBTABLE_HEADER*)MadtTable;
+    AcpiHeader->Length = sizeof(ACPI_TABLE_MADT);
+    TableEnd = (ULONG_PTR)MadtTable + MadtTable->Header.Length;
+
+    while ((ULONG_PTR)AcpiHeader <= TableEnd)
+    {
+        LocalApic = (ACPI_MADT_LOCAL_APIC*)AcpiHeader;
+
+        if (LocalApic->Header.Type == ACPI_MADT_TYPE_LOCAL_APIC && 
```

Comment:
I do recommend to strip the EOL space after the && or others will later hammer over your commit with sledgehammer-EOL-whitespace-removal-commits touching your file without any need and poisoning the history.